test/index.js

```diff
@@ -198,7 +198,15 @@ describe('with lazy loading', function() {
   });
 });
 
-describe('requiring from another directory', function() {
+describe('common functionality', function () {
+  it('throws a sensible error when not found', function () {
+    var x = gulpLoadPlugins({ config: __dirname + '/package.json' });
```

Comment:
The error only happens when we're passed a path (string) or undefined as the config, (L23 and 32). Passing in an object would cause the default `require` to be used (on L45), which already throws a helpful error.